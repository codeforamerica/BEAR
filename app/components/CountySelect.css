.countySelect {
  font-size: 2.3rem !important;
}
