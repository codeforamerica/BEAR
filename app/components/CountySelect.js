// @flow
import React, { Component } from 'react';

import counties from '../constants/californiaCounties';
import { FormCardContent } from './FormCard';

type Props = {
  onCountySelect: County => void
};

export default class Home extends Component<Props> {
  handleCountySelect = (e: SyntheticEvent<HTMLSelectElement>) => {
    const { onCountySelect } = this.props;
    onCountySelect({
      name: e.currentTarget.selectedOptions[0].text,
      code: e.currentTarget.value
    });
  };

  render() {
    return (
      <FormCardContent>
        <p className="form-question">Select your county</p>
        <div className="select">
          <select
            className="select__element"
            onChange={this.handleCountySelect}
            id="county-select"
            defaultValue=""
          >
            <option value="" disabled>
              Choose a county
            </option>
            {counties.map(county => (
              <option key={county} value={county.toUpperCase()}>
                {county}
              </option>
            ))}
          </select>
        </div>
      </FormCardContent>
    );
  }
}
