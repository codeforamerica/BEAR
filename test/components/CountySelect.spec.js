import sinon from 'sinon';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import CountySelect from '../../app/components/CountySelect';

Enzyme.configure({ adapter: new Adapter() });
const sandbox = sinon.createSandbox();

function setup() {
  const onCountySelectSpy = sandbox.spy();
  const component = shallow(
    <CountySelect onCountySelect={onCountySelectSpy} />
  );
  return {
    component,
    onCountySelectSpy
  };
}

afterEach(() => {
  sandbox.restore();
});

describe('CountySelect component', () => {
  it('selecting county should call onCountySelect with county name and value', () => {
    const { component, onCountySelectSpy } = setup();
    const countySelect = component.find('#county-select');
    const fakeEvent = {
      currentTarget: {
        value: 'CONTRA COSTA',
        selectedOptions: [{ text: 'Contra Costa' }]
      }
    };
    countySelect.simulate('change', fakeEvent);
    expect(onCountySelectSpy.called).toBe(true);
    expect(onCountySelectSpy.args[0][0]).toEqual({
      name: 'Contra Costa',
      code: 'CONTRA COSTA'
    });
  });

  it('should match exact snapshot', () => {
    const { component } = setup();
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
