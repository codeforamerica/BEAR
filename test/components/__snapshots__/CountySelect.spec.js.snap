// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountySelect component should match exact snapshot 1`] = `
<div>
  <p
    className="form-question"
  >
    Select your county
  </p>
  <div
    className="select"
  >
    <select
      className="select__element"
      id="county-select"
      onChange={[Function]}
    >
      <option
        disabled={true}
        value=""
      >
        Choose your county
      </option>
      <option
        value="ALAMEDA"
      >
        Alameda
      </option>
      <option
        value="ALPINE"
      >
        Alpine
      </option>
      <option
        value="AMADOR"
      >
        Amador
      </option>
      <option
        value="BUTTE"
      >
        Butte
      </option>
      <option
        value="CALAVERAS"
      >
        Calaveras
      </option>
      <option
        value="COLUSA"
      >
        Colusa
      </option>
      <option
        value="CONTRA COSTA"
      >
        Contra Costa
      </option>
      <option
        value="DEL NORTE"
      >
        Del Norte
      </option>
      <option
        value="EL DORADO"
      >
        El Dorado
      </option>
      <option
        value="FRESNO"
      >
        Fresno
      </option>
      <option
        value="GLENN"
      >
        Glenn
      </option>
      <option
        value="HUMBOLDT"
      >
        Humboldt
      </option>
      <option
        value="IMPERIAL"
      >
        Imperial
      </option>
      <option
        value="INYO"
      >
        Inyo
      </option>
      <option
        value="KERN"
      >
        Kern
      </option>
      <option
        value="KINGS"
      >
        Kings
      </option>
      <option
        value="LAKE"
      >
        Lake
      </option>
      <option
        value="LASSEN"
      >
        Lassen
      </option>
      <option
        value="LOS ANGELES"
      >
        Los Angeles
      </option>
      <option
        value="MADERA"
      >
        Madera
      </option>
      <option
        value="MARIN"
      >
        Marin
      </option>
      <option
        value="MARIPOSA"
      >
        Mariposa
      </option>
      <option
        value="MENDOCINO"
      >
        Mendocino
      </option>
      <option
        value="MERCED"
      >
        Merced
      </option>
      <option
        value="MODOC"
      >
        Modoc
      </option>
      <option
        value="MONO"
      >
        Mono
      </option>
      <option
        value="MONTEREY"
      >
        Monterey
      </option>
      <option
        value="NAPA"
      >
        Napa
      </option>
      <option
        value="NEVADA"
      >
        Nevada
      </option>
      <option
        value="ORANGE"
      >
        Orange
      </option>
      <option
        value="PLACER"
      >
        Placer
      </option>
      <option
        value="PLUMAS"
      >
        Plumas
      </option>
      <option
        value="RIVERSIDE"
      >
        Riverside
      </option>
      <option
        value="SACRAMENTO"
      >
        Sacramento
      </option>
      <option
        value="SAN BENITO"
      >
        San Benito
      </option>
      <option
        value="SAN BERNARDINO"
      >
        San Bernardino
      </option>
      <option
        value="SAN DIEGO"
      >
        San Diego
      </option>
      <option
        value="SAN FRANCISCO"
      >
        San Francisco
      </option>
      <option
        value="SAN JOAQUIN"
      >
        San Joaquin
      </option>
      <option
        value="SAN LUIS OBISPO"
      >
        San Luis Obispo
      </option>
      <option
        value="SAN MATEO"
      >
        San Mateo
      </option>
      <option
        value="SANTA BARBARA"
      >
        Santa Barbara
      </option>
      <option
        value="SANTA CLARA"
      >
        Santa Clara
      </option>
      <option
        value="SANTA CRUZ"
      >
        Santa Cruz
      </option>
      <option
        value="SHASTA"
      >
        Shasta
      </option>
      <option
        value="SIERRA"
      >
        Sierra
      </option>
      <option
        value="SISKIYOU"
      >
        Siskiyou
      </option>
      <option
        value="SOLANO"
      >
        Solano
      </option>
      <option
        value="SONOMA"
      >
        Sonoma
      </option>
      <option
        value="STANISLAUS"
      >
        Stanislaus
      </option>
      <option
        value="SUTTER"
      >
        Sutter
      </option>
      <option
        value="TEHAMA"
      >
        Tehama
      </option>
      <option
        value="TRINITY"
      >
        Trinity
      </option>
      <option
        value="TULARE"
      >
        Tulare
      </option>
      <option
        value="TUOLUMNE"
      >
        Tuolumne
      </option>
      <option
        value="VENTURA"
      >
        Ventura
      </option>
      <option
        value="YOLO"
      >
        Yolo
      </option>
      <option
        value="YUBA"
      >
        Yuba
      </option>
    </select>
  </div>
</div>
`;
